src/main/java/jenkins/plugins/git/GitToolChooser.java

```diff
@@ -86,16 +87,45 @@ public void addListener(TaskListener listener) {
      */
     private boolean decideAndUseCache(String remoteName) throws IOException, InterruptedException {
         boolean useCache = false;
-        String cacheEntry = AbstractGitSCMSource.getCacheEntry(remoteName);
-        File cacheDir = AbstractGitSCMSource.getCacheDir(cacheEntry, false);
-        if (cacheDir != null) {
-            Git git = Git.with(TaskListener.NULL, new EnvVars(EnvVars.masterEnvVars)).in(cacheDir).using("git");
-            GitClient client = git.getClient();
-            if (client.hasGitRepo()) {
-                sizeOfRepo = FileUtils.sizeOfDirectory(cacheDir);
-                sizeOfRepo = (sizeOfRepo/1000); // Conversion from Bytes to Kilo Bytes
-                useCache = true;
+        if (setSizeFromInternalCache(remoteName)) {
+            LOGGER.log(Level.INFO,
```

Comment:
Thanks very much for including the logging messages.  It makes the interactive evaluation much, much faster.

Messages a `INFO` level are written to the Jenkins log by default.  I'd strongly prefer that these diagnostic messages be at `FINE` level so that they default to not being written to the log.
```suggestion
            LOGGER.log(Level.FINE,
```